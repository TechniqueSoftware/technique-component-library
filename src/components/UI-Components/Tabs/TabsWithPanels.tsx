import * as React from 'react';
import { Tabs } from './Tabs';
import { Tab } from './Tab';
import Typography from '@material-ui/core/Typography/Typography';
import Paper from '../Paper/Paper';

const { useState, useEffect } = React;
type TabAndPanelProps = { id: string, label: string, renderContent: (isActiveTab: boolean) => React.ReactNode };

export function TabPanelContainer(props: { id: string, isActive: boolean, children: React.ReactNode }) {
  const { id, isActive, children } = props;
  const hidden = !isActive;
  return <Typography
      component="div"
      role="tabpanel"
      hidden={hidden}
      id={`panel-${id}`}
      aria-labelledby={`panel-${id}`}
    >
      {children}
  </Typography>;
}

export type TabsWithPanelsProps = {
  tabAndPanelPropsList: TabAndPanelProps[],
  tabsAriaLabel: string,
  initialTab?: string;
  onTabChange?: (tabId: string) => void;
};

export function TabsWithPanels(props: TabsWithPanelsProps) {

  const {
    tabAndPanelPropsList,
    tabsAriaLabel,
    initialTab,
    onTabChange
  } = props;

  const getInitialTab = () => initialTab || tabAndPanelPropsList[0] && tabAndPanelPropsList[0].id || '';

  const [currentTab, setCurrentTab] = useState<string>(getInitialTab);
  const [initializedTabs, setInitializedTabs] = useState<Set<string>>(new Set());
  useEffect(() => {
    initializedTabs.add(currentTab);
    setInitializedTabs(initializedTabs);
  }, [currentTab]);

  function handleTabChange(event: any, value: string) {
    setCurrentTab(value);
    // Report the tab change to the parent component in case it needs to do something based on the change like update the queryParams to
    // represent the current location.
    if (onTabChange) {
      onTabChange(value);
    }
  }

  /**
   * We only want to render the currentTab or any tabs that have been rendered previously.
   */
  function renderPanelContentIfInitialized(panelContentList: TabAndPanelProps[], activeTabId: string) {
    return panelContentList.map((props) => {
      const { id, renderContent } = props;
      const isCurrentTab = activeTabId === id;
      if (isCurrentTab || initializedTabs.has(id)) {
        return <TabPanelContainer isActive={isCurrentTab} id={id} key={id}>
          {renderContent(isCurrentTab)}
        </TabPanelContainer>;
      }
    });
  }

  function renderTabs(tabPropsList: { id: string, label: string }[]) {
    return tabPropsList.map((props: any) => {
      const { id, label } = props;
      const tabId = `tab-${id}`;
      const tabProps = {
        label,
        id: tabId,
        key: tabId,
        'aria-controls': id,
        value: id
      };
      return <Tab {...tabProps} />;
    });
  }

  return (
    <Paper>
        <Tabs
          value={currentTab}
          onChange={handleTabChange}
          aria-label={tabsAriaLabel}
          // Using useStyles to apply this style seemed to cause a bug where not all the tabs would display. Using inline styles seems
          // to work fine. I don't think it's worth the time to debug the issue.
          style={{ borderBottom: '1px solid #ddd' }}
        >
          {renderTabs(tabAndPanelPropsList)}
        </Tabs>
      {renderPanelContentIfInitialized(tabAndPanelPropsList, currentTab)}
    </Paper>);
}
