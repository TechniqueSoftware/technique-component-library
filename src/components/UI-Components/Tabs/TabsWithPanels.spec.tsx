import * as React from 'react';
import { mount } from 'enzyme';
import { TabsWithPanels, TabsWithPanelsProps } from './TabsWithPanels';
import { act } from 'react-dom/test-utils';

describe('TabsWithPanels', () => {

  let baseProps: TabsWithPanelsProps;
  beforeEach(() => {
    baseProps = {
      initialTab: '1',
      onTabChange: jest.fn(),
      tabAndPanelPropsList: [
        { renderContent: () => <span>mockTab1Content</span>, id: '1', label: 'mockTab1Label' },
        { renderContent: () => <span>mockTab2Content</span>, id: '2', label: 'mockTab1Label' },
        { renderContent: () => <span>mockTab3Content</span>, id: '3', label: 'mockTab1Label' }
      ],
      tabsAriaLabel: 'mockTestAriaLabel'
    };
  });

  test('should render with minimal props', () => {
    const props: TabsWithPanelsProps = { ...baseProps };
    const reactWrapper = mount(<TabsWithPanels {...props} />);
    const tabPanelContainers = reactWrapper.find('TabPanelContainer');
    // Only one tabPanelContainer should render initially.
    expect(tabPanelContainers.length).toBe(1);
    expect(reactWrapper.debug()).toMatchSnapshot();
  });

  test('should render a new content when clicking on tab', async () => {
    const props: TabsWithPanelsProps = { ...baseProps };
    const reactWrapper = mount(<TabsWithPanels {...props} />);
    const tab1Elements = reactWrapper.find('#tab-2').first();
    await act(async () => {
      tab1Elements.simulate('click');
    });

    reactWrapper.update();

    const tabPanelContainers = reactWrapper.find('TabPanelContainer');
    // Once a tab has done it's initial render it should still be rendered but not visible.
    expect(tabPanelContainers.length).toBe(2);
    expect(reactWrapper.debug()).toMatchSnapshot();
  });

});
