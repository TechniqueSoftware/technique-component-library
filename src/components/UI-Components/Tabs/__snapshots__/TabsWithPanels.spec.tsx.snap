// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabsWithPanels should render a new content when clicking on tab 1`] = `
"<TabsWithPanels initialTab=\\"1\\" onTabChange={[Function: mockConstructor] { _isMockFunction: true, getMockImplementation: [Function (anonymous)], mock: Object [Object: null prototype] { calls: [ [ '2' ] ], instances: [ undefined ], invocationCallOrder: [ 1 ], results: [ Object [Object: null prototype] { isThrow: false, value: undefined } ] }, mockClear: [Function (anonymous)], mockReset: [Function (anonymous)], mockRestore: [Function (anonymous)], mockReturnValueOnce: [Function (anonymous)], mockResolvedValueOnce: [Function (anonymous)], mockRejectedValueOnce: [Function (anonymous)], mockReturnValue: [Function (anonymous)], mockResolvedValue: [Function (anonymous)], mockRejectedValue: [Function (anonymous)], mockImplementationOnce: [Function (anonymous)], mockImplementation: [Function (anonymous)], mockReturnThis: [Function (anonymous)], mockName: [Function (anonymous)], getMockName: [Function (anonymous)] }} tabAndPanelPropsList={{...}} tabsAriaLabel=\\"mockTestAriaLabel\\">
  <WithStyles(ForwardRef(Paper))>
    <ForwardRef(Paper) classes={{...}}>
      <div className=\\"MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded\\">
        <WithStyles(ForwardRef(Tabs)) value=\\"2\\" onChange={[Function: handleTabChange]} aria-label=\\"mockTestAriaLabel\\" style={{...}}>
          <ForwardRef(Tabs) classes={{...}} value=\\"2\\" onChange={[Function: handleTabChange]} aria-label=\\"mockTestAriaLabel\\" style={{...}}>
            <div className=\\"MuiTabs-root\\" style={{...}}>
              <div className=\\"MuiTabs-scroller MuiTabs-fixed\\" style={{...}} onScroll={[Function: debounced] { clear: [Function (anonymous)] }}>
                <div aria-label=\\"mockTestAriaLabel\\" aria-labelledby={[undefined]} className=\\"MuiTabs-flexContainer\\" onKeyDown={[Function: handleKeyDown]} role=\\"tablist\\">
                  <WithStyles(ForwardRef(Tab)) label=\\"mockTab1Label\\" id=\\"tab-1\\" aria-controls=\\"1\\" value=\\"1\\" fullWidth={false} indicator={false} selected={false} selectionFollowsFocus={[undefined]} onChange={[Function: handleTabChange]} textColor=\\"inherit\\">
                    <ForwardRef(Tab) classes={{...}} label=\\"mockTab1Label\\" id=\\"tab-1\\" aria-controls=\\"1\\" value=\\"1\\" fullWidth={false} indicator={false} selected={false} selectionFollowsFocus={[undefined]} onChange={[Function: handleTabChange]} textColor=\\"inherit\\">
                      <WithStyles(ForwardRef(ButtonBase)) focusRipple={true} className=\\"MuiTab-root MuiTab-textColorInherit\\" role=\\"tab\\" aria-selected={false} disabled={false} onClick={[Function: handleClick]} onFocus={[Function: handleFocus]} tabIndex={-1} id=\\"tab-1\\" aria-controls=\\"1\\">
                        <ForwardRef(ButtonBase) classes={{...}} focusRipple={true} className=\\"MuiTab-root MuiTab-textColorInherit\\" role=\\"tab\\" aria-selected={false} disabled={false} onClick={[Function: handleClick]} onFocus={[Function: handleFocus]} tabIndex={-1} id=\\"tab-1\\" aria-controls=\\"1\\">
                          <button className=\\"MuiButtonBase-root MuiTab-root MuiTab-textColorInherit\\" onBlur={[Function (anonymous)]} onClick={[Function: handleClick]} onFocus={[Function (anonymous)]} onKeyDown={[Function (anonymous)]} onKeyUp={[Function (anonymous)]} onMouseDown={[Function (anonymous)]} onMouseLeave={[Function (anonymous)]} onMouseUp={[Function (anonymous)]} onDragLeave={[Function (anonymous)]} onTouchEnd={[Function (anonymous)]} onTouchMove={[Function (anonymous)]} onTouchStart={[Function (anonymous)]} tabIndex={-1} type=\\"button\\" disabled={false} role=\\"tab\\" aria-selected={false} id=\\"tab-1\\" aria-controls=\\"1\\">
                            <span className=\\"MuiTab-wrapper\\">
                              mockTab1Label
                            </span>
                            <WithStyles(memo) center={false}>
                              <ForwardRef(TouchRipple) classes={{...}} center={false}>
                                <span className=\\"MuiTouchRipple-root\\">
                                  <TransitionGroup component={{...}} exit={true} childFactory={[Function: childFactory]} />
                                </span>
                              </ForwardRef(TouchRipple)>
                            </WithStyles(memo)>
                          </button>
                        </ForwardRef(ButtonBase)>
                      </WithStyles(ForwardRef(ButtonBase))>
                    </ForwardRef(Tab)>
                  </WithStyles(ForwardRef(Tab))>
                  <WithStyles(ForwardRef(Tab)) label=\\"mockTab1Label\\" id=\\"tab-2\\" aria-controls=\\"2\\" value=\\"2\\" fullWidth={false} indicator={false} selected={true} selectionFollowsFocus={[undefined]} onChange={[Function: handleTabChange]} textColor=\\"inherit\\">
                    <ForwardRef(Tab) classes={{...}} label=\\"mockTab1Label\\" id=\\"tab-2\\" aria-controls=\\"2\\" value=\\"2\\" fullWidth={false} indicator={false} selected={true} selectionFollowsFocus={[undefined]} onChange={[Function: handleTabChange]} textColor=\\"inherit\\">
                      <WithStyles(ForwardRef(ButtonBase)) focusRipple={true} className=\\"MuiTab-root MuiTab-textColorInherit Mui-selected\\" role=\\"tab\\" aria-selected={true} disabled={false} onClick={[Function: handleClick]} onFocus={[Function: handleFocus]} tabIndex={0} id=\\"tab-2\\" aria-controls=\\"2\\">
                        <ForwardRef(ButtonBase) classes={{...}} focusRipple={true} className=\\"MuiTab-root MuiTab-textColorInherit Mui-selected\\" role=\\"tab\\" aria-selected={true} disabled={false} onClick={[Function: handleClick]} onFocus={[Function: handleFocus]} tabIndex={0} id=\\"tab-2\\" aria-controls=\\"2\\">
                          <button className=\\"MuiButtonBase-root MuiTab-root MuiTab-textColorInherit Mui-selected\\" onBlur={[Function (anonymous)]} onClick={[Function: handleClick]} onFocus={[Function (anonymous)]} onKeyDown={[Function (anonymous)]} onKeyUp={[Function (anonymous)]} onMouseDown={[Function (anonymous)]} onMouseLeave={[Function (anonymous)]} onMouseUp={[Function (anonymous)]} onDragLeave={[Function (anonymous)]} onTouchEnd={[Function (anonymous)]} onTouchMove={[Function (anonymous)]} onTouchStart={[Function (anonymous)]} tabIndex={0} type=\\"button\\" disabled={false} role=\\"tab\\" aria-selected={true} id=\\"tab-2\\" aria-controls=\\"2\\">
                            <span className=\\"MuiTab-wrapper\\">
                              mockTab1Label
                            </span>
                            <WithStyles(memo) center={false}>
                              <ForwardRef(TouchRipple) classes={{...}} center={false}>
                                <span className=\\"MuiTouchRipple-root\\">
                                  <TransitionGroup component={{...}} exit={true} childFactory={[Function: childFactory]} />
                                </span>
                              </ForwardRef(TouchRipple)>
                            </WithStyles(memo)>
                          </button>
                        </ForwardRef(ButtonBase)>
                      </WithStyles(ForwardRef(ButtonBase))>
                    </ForwardRef(Tab)>
                  </WithStyles(ForwardRef(Tab))>
                  <WithStyles(ForwardRef(Tab)) label=\\"mockTab1Label\\" id=\\"tab-3\\" aria-controls=\\"3\\" value=\\"3\\" fullWidth={false} indicator={false} selected={false} selectionFollowsFocus={[undefined]} onChange={[Function: handleTabChange]} textColor=\\"inherit\\">
                    <ForwardRef(Tab) classes={{...}} label=\\"mockTab1Label\\" id=\\"tab-3\\" aria-controls=\\"3\\" value=\\"3\\" fullWidth={false} indicator={false} selected={false} selectionFollowsFocus={[undefined]} onChange={[Function: handleTabChange]} textColor=\\"inherit\\">
                      <WithStyles(ForwardRef(ButtonBase)) focusRipple={true} className=\\"MuiTab-root MuiTab-textColorInherit\\" role=\\"tab\\" aria-selected={false} disabled={false} onClick={[Function: handleClick]} onFocus={[Function: handleFocus]} tabIndex={-1} id=\\"tab-3\\" aria-controls=\\"3\\">
                        <ForwardRef(ButtonBase) classes={{...}} focusRipple={true} className=\\"MuiTab-root MuiTab-textColorInherit\\" role=\\"tab\\" aria-selected={false} disabled={false} onClick={[Function: handleClick]} onFocus={[Function: handleFocus]} tabIndex={-1} id=\\"tab-3\\" aria-controls=\\"3\\">
                          <button className=\\"MuiButtonBase-root MuiTab-root MuiTab-textColorInherit\\" onBlur={[Function (anonymous)]} onClick={[Function: handleClick]} onFocus={[Function (anonymous)]} onKeyDown={[Function (anonymous)]} onKeyUp={[Function (anonymous)]} onMouseDown={[Function (anonymous)]} onMouseLeave={[Function (anonymous)]} onMouseUp={[Function (anonymous)]} onDragLeave={[Function (anonymous)]} onTouchEnd={[Function (anonymous)]} onTouchMove={[Function (anonymous)]} onTouchStart={[Function (anonymous)]} tabIndex={-1} type=\\"button\\" disabled={false} role=\\"tab\\" aria-selected={false} id=\\"tab-3\\" aria-controls=\\"3\\">
                            <span className=\\"MuiTab-wrapper\\">
                              mockTab1Label
                            </span>
                            <WithStyles(memo) center={false}>
                              <ForwardRef(TouchRipple) classes={{...}} center={false}>
                                <span className=\\"MuiTouchRipple-root\\">
                                  <TransitionGroup component={{...}} exit={true} childFactory={[Function: childFactory]} />
                                </span>
                              </ForwardRef(TouchRipple)>
                            </WithStyles(memo)>
                          </button>
                        </ForwardRef(ButtonBase)>
                      </WithStyles(ForwardRef(ButtonBase))>
                    </ForwardRef(Tab)>
                  </WithStyles(ForwardRef(Tab))>
                </div>
                <WithStyles(ForwardRef(TabIndicator)) className=\\"MuiTabs-indicator\\" orientation=\\"horizontal\\" color=\\"secondary\\" style={{...}}>
                  <ForwardRef(TabIndicator) classes={{...}} className=\\"MuiTabs-indicator\\" orientation=\\"horizontal\\" color=\\"secondary\\" style={{...}}>
                    <span className=\\"PrivateTabIndicator-root-1 PrivateTabIndicator-colorSecondary-3 MuiTabs-indicator\\" style={{...}} />
                  </ForwardRef(TabIndicator)>
                </WithStyles(ForwardRef(TabIndicator))>
              </div>
            </div>
          </ForwardRef(Tabs)>
        </WithStyles(ForwardRef(Tabs))>
        <TabPanelContainer isActive={false} id=\\"1\\">
          <WithStyles(ForwardRef(Typography)) component=\\"div\\" role=\\"tabpanel\\" hidden={true} id=\\"panel-1\\" aria-labelledby=\\"panel-1\\">
            <ForwardRef(Typography) classes={{...}} component=\\"div\\" role=\\"tabpanel\\" hidden={true} id=\\"panel-1\\" aria-labelledby=\\"panel-1\\">
              <div className=\\"MuiTypography-root MuiTypography-body1\\" role=\\"tabpanel\\" hidden={true} id=\\"panel-1\\" aria-labelledby=\\"panel-1\\">
                <span>
                  mockTab1Content
                </span>
              </div>
            </ForwardRef(Typography)>
          </WithStyles(ForwardRef(Typography))>
        </TabPanelContainer>
        <TabPanelContainer isActive={true} id=\\"2\\">
          <WithStyles(ForwardRef(Typography)) component=\\"div\\" role=\\"tabpanel\\" hidden={false} id=\\"panel-2\\" aria-labelledby=\\"panel-2\\">
            <ForwardRef(Typography) classes={{...}} component=\\"div\\" role=\\"tabpanel\\" hidden={false} id=\\"panel-2\\" aria-labelledby=\\"panel-2\\">
              <div className=\\"MuiTypography-root MuiTypography-body1\\" role=\\"tabpanel\\" hidden={false} id=\\"panel-2\\" aria-labelledby=\\"panel-2\\">
                <span>
                  mockTab2Content
                </span>
              </div>
            </ForwardRef(Typography)>
          </WithStyles(ForwardRef(Typography))>
        </TabPanelContainer>
      </div>
    </ForwardRef(Paper)>
  </WithStyles(ForwardRef(Paper))>
</TabsWithPanels>"
`;

exports[`TabsWithPanels should render with minimal props 1`] = `
"<TabsWithPanels initialTab=\\"1\\" onTabChange={[Function: mockConstructor] { _isMockFunction: true, getMockImplementation: [Function (anonymous)], mock: Object [Object: null prototype] { calls: [], instances: [], invocationCallOrder: [], results: [] }, mockClear: [Function (anonymous)], mockReset: [Function (anonymous)], mockRestore: [Function (anonymous)], mockReturnValueOnce: [Function (anonymous)], mockResolvedValueOnce: [Function (anonymous)], mockRejectedValueOnce: [Function (anonymous)], mockReturnValue: [Function (anonymous)], mockResolvedValue: [Function (anonymous)], mockRejectedValue: [Function (anonymous)], mockImplementationOnce: [Function (anonymous)], mockImplementation: [Function (anonymous)], mockReturnThis: [Function (anonymous)], mockName: [Function (anonymous)], getMockName: [Function (anonymous)] }} tabAndPanelPropsList={{...}} tabsAriaLabel=\\"mockTestAriaLabel\\">
  <WithStyles(ForwardRef(Paper))>
    <ForwardRef(Paper) classes={{...}}>
      <div className=\\"MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded\\">
        <WithStyles(ForwardRef(Tabs)) value=\\"1\\" onChange={[Function: handleTabChange]} aria-label=\\"mockTestAriaLabel\\" style={{...}}>
          <ForwardRef(Tabs) classes={{...}} value=\\"1\\" onChange={[Function: handleTabChange]} aria-label=\\"mockTestAriaLabel\\" style={{...}}>
            <div className=\\"MuiTabs-root\\" style={{...}}>
              <div className=\\"MuiTabs-scroller MuiTabs-fixed\\" style={{...}} onScroll={[Function: debounced] { clear: [Function (anonymous)] }}>
                <div aria-label=\\"mockTestAriaLabel\\" aria-labelledby={[undefined]} className=\\"MuiTabs-flexContainer\\" onKeyDown={[Function: handleKeyDown]} role=\\"tablist\\">
                  <WithStyles(ForwardRef(Tab)) label=\\"mockTab1Label\\" id=\\"tab-1\\" aria-controls=\\"1\\" value=\\"1\\" fullWidth={false} indicator={false} selected={true} selectionFollowsFocus={[undefined]} onChange={[Function: handleTabChange]} textColor=\\"inherit\\">
                    <ForwardRef(Tab) classes={{...}} label=\\"mockTab1Label\\" id=\\"tab-1\\" aria-controls=\\"1\\" value=\\"1\\" fullWidth={false} indicator={false} selected={true} selectionFollowsFocus={[undefined]} onChange={[Function: handleTabChange]} textColor=\\"inherit\\">
                      <WithStyles(ForwardRef(ButtonBase)) focusRipple={true} className=\\"MuiTab-root MuiTab-textColorInherit Mui-selected\\" role=\\"tab\\" aria-selected={true} disabled={false} onClick={[Function: handleClick]} onFocus={[Function: handleFocus]} tabIndex={0} id=\\"tab-1\\" aria-controls=\\"1\\">
                        <ForwardRef(ButtonBase) classes={{...}} focusRipple={true} className=\\"MuiTab-root MuiTab-textColorInherit Mui-selected\\" role=\\"tab\\" aria-selected={true} disabled={false} onClick={[Function: handleClick]} onFocus={[Function: handleFocus]} tabIndex={0} id=\\"tab-1\\" aria-controls=\\"1\\">
                          <button className=\\"MuiButtonBase-root MuiTab-root MuiTab-textColorInherit Mui-selected\\" onBlur={[Function (anonymous)]} onClick={[Function: handleClick]} onFocus={[Function (anonymous)]} onKeyDown={[Function (anonymous)]} onKeyUp={[Function (anonymous)]} onMouseDown={[Function (anonymous)]} onMouseLeave={[Function (anonymous)]} onMouseUp={[Function (anonymous)]} onDragLeave={[Function (anonymous)]} onTouchEnd={[Function (anonymous)]} onTouchMove={[Function (anonymous)]} onTouchStart={[Function (anonymous)]} tabIndex={0} type=\\"button\\" disabled={false} role=\\"tab\\" aria-selected={true} id=\\"tab-1\\" aria-controls=\\"1\\">
                            <span className=\\"MuiTab-wrapper\\">
                              mockTab1Label
                            </span>
                            <WithStyles(memo) center={false}>
                              <ForwardRef(TouchRipple) classes={{...}} center={false}>
                                <span className=\\"MuiTouchRipple-root\\">
                                  <TransitionGroup component={{...}} exit={true} childFactory={[Function: childFactory]} />
                                </span>
                              </ForwardRef(TouchRipple)>
                            </WithStyles(memo)>
                          </button>
                        </ForwardRef(ButtonBase)>
                      </WithStyles(ForwardRef(ButtonBase))>
                    </ForwardRef(Tab)>
                  </WithStyles(ForwardRef(Tab))>
                  <WithStyles(ForwardRef(Tab)) label=\\"mockTab1Label\\" id=\\"tab-2\\" aria-controls=\\"2\\" value=\\"2\\" fullWidth={false} indicator={false} selected={false} selectionFollowsFocus={[undefined]} onChange={[Function: handleTabChange]} textColor=\\"inherit\\">
                    <ForwardRef(Tab) classes={{...}} label=\\"mockTab1Label\\" id=\\"tab-2\\" aria-controls=\\"2\\" value=\\"2\\" fullWidth={false} indicator={false} selected={false} selectionFollowsFocus={[undefined]} onChange={[Function: handleTabChange]} textColor=\\"inherit\\">
                      <WithStyles(ForwardRef(ButtonBase)) focusRipple={true} className=\\"MuiTab-root MuiTab-textColorInherit\\" role=\\"tab\\" aria-selected={false} disabled={false} onClick={[Function: handleClick]} onFocus={[Function: handleFocus]} tabIndex={-1} id=\\"tab-2\\" aria-controls=\\"2\\">
                        <ForwardRef(ButtonBase) classes={{...}} focusRipple={true} className=\\"MuiTab-root MuiTab-textColorInherit\\" role=\\"tab\\" aria-selected={false} disabled={false} onClick={[Function: handleClick]} onFocus={[Function: handleFocus]} tabIndex={-1} id=\\"tab-2\\" aria-controls=\\"2\\">
                          <button className=\\"MuiButtonBase-root MuiTab-root MuiTab-textColorInherit\\" onBlur={[Function (anonymous)]} onClick={[Function: handleClick]} onFocus={[Function (anonymous)]} onKeyDown={[Function (anonymous)]} onKeyUp={[Function (anonymous)]} onMouseDown={[Function (anonymous)]} onMouseLeave={[Function (anonymous)]} onMouseUp={[Function (anonymous)]} onDragLeave={[Function (anonymous)]} onTouchEnd={[Function (anonymous)]} onTouchMove={[Function (anonymous)]} onTouchStart={[Function (anonymous)]} tabIndex={-1} type=\\"button\\" disabled={false} role=\\"tab\\" aria-selected={false} id=\\"tab-2\\" aria-controls=\\"2\\">
                            <span className=\\"MuiTab-wrapper\\">
                              mockTab1Label
                            </span>
                            <WithStyles(memo) center={false}>
                              <ForwardRef(TouchRipple) classes={{...}} center={false}>
                                <span className=\\"MuiTouchRipple-root\\">
                                  <TransitionGroup component={{...}} exit={true} childFactory={[Function: childFactory]} />
                                </span>
                              </ForwardRef(TouchRipple)>
                            </WithStyles(memo)>
                          </button>
                        </ForwardRef(ButtonBase)>
                      </WithStyles(ForwardRef(ButtonBase))>
                    </ForwardRef(Tab)>
                  </WithStyles(ForwardRef(Tab))>
                  <WithStyles(ForwardRef(Tab)) label=\\"mockTab1Label\\" id=\\"tab-3\\" aria-controls=\\"3\\" value=\\"3\\" fullWidth={false} indicator={false} selected={false} selectionFollowsFocus={[undefined]} onChange={[Function: handleTabChange]} textColor=\\"inherit\\">
                    <ForwardRef(Tab) classes={{...}} label=\\"mockTab1Label\\" id=\\"tab-3\\" aria-controls=\\"3\\" value=\\"3\\" fullWidth={false} indicator={false} selected={false} selectionFollowsFocus={[undefined]} onChange={[Function: handleTabChange]} textColor=\\"inherit\\">
                      <WithStyles(ForwardRef(ButtonBase)) focusRipple={true} className=\\"MuiTab-root MuiTab-textColorInherit\\" role=\\"tab\\" aria-selected={false} disabled={false} onClick={[Function: handleClick]} onFocus={[Function: handleFocus]} tabIndex={-1} id=\\"tab-3\\" aria-controls=\\"3\\">
                        <ForwardRef(ButtonBase) classes={{...}} focusRipple={true} className=\\"MuiTab-root MuiTab-textColorInherit\\" role=\\"tab\\" aria-selected={false} disabled={false} onClick={[Function: handleClick]} onFocus={[Function: handleFocus]} tabIndex={-1} id=\\"tab-3\\" aria-controls=\\"3\\">
                          <button className=\\"MuiButtonBase-root MuiTab-root MuiTab-textColorInherit\\" onBlur={[Function (anonymous)]} onClick={[Function: handleClick]} onFocus={[Function (anonymous)]} onKeyDown={[Function (anonymous)]} onKeyUp={[Function (anonymous)]} onMouseDown={[Function (anonymous)]} onMouseLeave={[Function (anonymous)]} onMouseUp={[Function (anonymous)]} onDragLeave={[Function (anonymous)]} onTouchEnd={[Function (anonymous)]} onTouchMove={[Function (anonymous)]} onTouchStart={[Function (anonymous)]} tabIndex={-1} type=\\"button\\" disabled={false} role=\\"tab\\" aria-selected={false} id=\\"tab-3\\" aria-controls=\\"3\\">
                            <span className=\\"MuiTab-wrapper\\">
                              mockTab1Label
                            </span>
                            <WithStyles(memo) center={false}>
                              <ForwardRef(TouchRipple) classes={{...}} center={false}>
                                <span className=\\"MuiTouchRipple-root\\">
                                  <TransitionGroup component={{...}} exit={true} childFactory={[Function: childFactory]} />
                                </span>
                              </ForwardRef(TouchRipple)>
                            </WithStyles(memo)>
                          </button>
                        </ForwardRef(ButtonBase)>
                      </WithStyles(ForwardRef(ButtonBase))>
                    </ForwardRef(Tab)>
                  </WithStyles(ForwardRef(Tab))>
                </div>
                <WithStyles(ForwardRef(TabIndicator)) className=\\"MuiTabs-indicator\\" orientation=\\"horizontal\\" color=\\"secondary\\" style={{...}}>
                  <ForwardRef(TabIndicator) classes={{...}} className=\\"MuiTabs-indicator\\" orientation=\\"horizontal\\" color=\\"secondary\\" style={{...}}>
                    <span className=\\"PrivateTabIndicator-root-1 PrivateTabIndicator-colorSecondary-3 MuiTabs-indicator\\" style={{...}} />
                  </ForwardRef(TabIndicator)>
                </WithStyles(ForwardRef(TabIndicator))>
              </div>
            </div>
          </ForwardRef(Tabs)>
        </WithStyles(ForwardRef(Tabs))>
        <TabPanelContainer isActive={true} id=\\"1\\">
          <WithStyles(ForwardRef(Typography)) component=\\"div\\" role=\\"tabpanel\\" hidden={false} id=\\"panel-1\\" aria-labelledby=\\"panel-1\\">
            <ForwardRef(Typography) classes={{...}} component=\\"div\\" role=\\"tabpanel\\" hidden={false} id=\\"panel-1\\" aria-labelledby=\\"panel-1\\">
              <div className=\\"MuiTypography-root MuiTypography-body1\\" role=\\"tabpanel\\" hidden={false} id=\\"panel-1\\" aria-labelledby=\\"panel-1\\">
                <span>
                  mockTab1Content
                </span>
              </div>
            </ForwardRef(Typography)>
          </WithStyles(ForwardRef(Typography))>
        </TabPanelContainer>
      </div>
    </ForwardRef(Paper)>
  </WithStyles(ForwardRef(Paper))>
</TabsWithPanels>"
`;
